### PR TITLE
Added the smos-convert executable

### PR DESCRIPTION
**Pull request by CSVdB**

### CODE REVIEW

File: smos-convert/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.stack-work/
+smos-convert.cabal
+*~
```
Comment (NorfairKing) [inline smos-convert/.gitignore]: add a newline.

File: smos-convert/LICENSE [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Copyright Author name here (c) 2018
```
Comment (NorfairKing) [inline smos-convert/LICENSE]: make this 'all rights reserved'  for now.

File: smos-convert/app/Main.hs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module Main where
+
+import Convert
+
+main :: IO ()
+main = mainFunc
```
Comment (NorfairKing) [inline smos-convert/app/Main.hs]: `smosConvert` instead of `mainFunc`

File: smos-convert/app/Main.hs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module Main where
+
+import Convert
```
Comment (NorfairKing) [inline smos-convert/app/Main.hs]: `Smos.Convert` instead of `Convert`

File: smos-convert/package.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
+name:                smos-convert
+version:             0.1.0.0
+github:              "githubuser/smos-convert"
+license:             BSD3
+author:              "Author name here"
+maintainer:          "example@example.com"
+copyright:           "2018 Author name here"
```
Comment (NorfairKing) [inline smos-convert/package.yaml]: fill in all the above data correctly

File: smos-convert/package.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+- ChangeLog.md
+
+# Metadata used when publishing your package
+# synopsis:            Short description of your package
+# category:            Web
+
+# To avoid duplicated efforts in documentation and dealing with the
+# complications of embedding Haddock markup inside cabal files, it is
+# common to point users to the README.md file.
+description:         Please see the README on Github at <https://github.com/githubuser/smos-convert#readme>
```
Comment (NorfairKing) [inline smos-convert/package.yaml]: the right link, please.

File: smos-data/src/Smos/Data/Types.hs [hunk 7]
```diff
@@ -126,7 +126,8 @@ instance FromJSON Entry where
 
 instance ToJSON Entry where
     toJSON Entry {..} =
-        if and [ isNothing entryContents
+        if and
```
Comment (NorfairKing) [inline smos-data/src/Smos/Data/Types.hs:129]: Don't make these cosmetic changes here.

File: smos/src/Smos/Draw.hs [hunk 8]
```diff
@@ -55,7 +55,8 @@ drawNoContent :: Widget n
 drawNoContent =
     B.vCenterLayer $
     B.vBox $
-    map B.hCenterLayer
+    map
```
Comment (NorfairKing) [inline smos/src/Smos/Draw.hs:58]: idem

**Review by NorfairKing — CHANGES_REQUESTED** [review]

File: smos-data/src/Smos/Data/Types.hs (hunk 7 above)
Comment (CSVdB) [reply smos-data/src/Smos/Data/Types.hs:129]: I didn't change anything about the code in this file, zift changed this... We need to sync up the config zift uses.

Comment (NorfairKing) [reply smos-data/src/Smos/Data/Types.hs:129]: You can still un-add just these parts.

File: smos-convert/.gitignore [hunk 9]
```diff
@@ -1,3 +1,4 @@
 .stack-work/
 smos-convert.cabal
-*~
\ No newline at end of file
+*~
```
Comment (NorfairKing) [inline smos-convert/.gitignore]: Just one was enough.

File: smos-convert/LICENSE [hunk 10]
```diff
@@ -1,5 +1,3 @@
-Copyright Author name here (c) 2018
-
```
Comment (NorfairKing) [inline smos-convert/LICENSE]: You still need the copyright notice with the author's name and date, but not the BSD license.

So you've done exactly the opposite of what we need.

**Review by NorfairKing — CHANGES_REQUESTED** [review]

**Pull request closed without merging**